src/Command/UpdateSimpleProducts.php

```diff
@@ -63,42 +81,101 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        $limit = $input->getOption('limit');
+
         $this->manager->dropAndCreateIndex();
 
-        $syliusProducts = $this->syliusProductRepository->findAll();
-        $syliusSimpleProducts = array_filter($syliusProducts, function (ProductInterface $product) {
-            return $product->isSimple();
-        });
+        /** @var Channelinterface[] $channels */
+        $channels = $this->syliusChannelRepository->findAll();
 
-        $productDocumentCreatedCount = 0;
-        /** @var ProductInterface[] $syliusSimpleProducts */
-        foreach ($syliusSimpleProducts as $syliusProduct) {
-            $channels = $syliusProduct->getChannels();
+        foreach ($channels as $channel) {
+            $locales = $channel->getLocales();
```

Comment:
Body of this for each could go to separate private class, just to follow object calisthenics